addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -26,12 +26,16 @@ def _compute_is_forward_operator(self):
         self.filtered(lambda s: s.step_type == "create_lead_and_forward").is_forward_operator = True
 
     def _chatbot_crm_prepare_lead_values(self, discuss_channel, description):
+        partner = self.env.user.partner_id
+        team = self.crm_team_id
+        if partner.company_id and team.company_id and partner.company_id != team.company_id:
+            team = self.env["crm.team"]
         return {
-            "company_id": self.crm_team_id.company_id.id,
+            "company_id": team.company_id.id or partner.company_id.id,
```

Comment:
https://github.com/odoo/odoo/pull/190364#discussion_r1903912576

Did not take long. Set to False, we generally don't want a forced company in automatically generated leads.

Reply:
Alternatively let the compute do its job, and if there is an issue maybe the compute should be updated.

Reply:
I guess the compute should be updated then because if I set it to `False` the tests (or actual flow) won't pass.

If the partner has a company the only way to assign the lead without a crash is to have the same company on the lead.

If the team has a company the only way to assign the lead without a crash is to have the same company on the lead.

Reply:
- “Testing Bot's New Lead” belongs to company “False” and “Sales Team” (team_id: 'Test Sale Team 2') belongs to another company.

Reply:
Ok I got it, the compute was not called because another line (not in previous diff) was also assigning the company.
Now I removed both of them and only let the compute work, which seems to be fine after all.